### PR TITLE
policy list should work after associated hg removal

### PR DESCRIPTION
**Pull request by pondrejk**
### Problem Statement
SAT-19502 describes the issue, https://github.com/theforeman/foreman_openscap/pull/568/files provides the upstream fix

### Solution
Extending an existing test to check the policy behavior after hostgroup removal

### CODE REVIEW

**Comment by pondrejk** [comment]
```
"trigger": "test-robottelo"
"pytest": "tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups"
"theforeman":
    "foreman_openscap": "568"
```

**Comment by pondrejk** [comment]
Disregard the prt run, the test collection there didn't match my comment, so I killed the job, investigating...

**Comment by Griffin-Sullivan** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups
theforeman:
    foreman_openscap: 568
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6879
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups --external-logging
Test Result : ========== 1 passed, 70 deselected, 79 warnings in 1633.87s (0:27:13) ==========
```

File: tests/foreman/cli/test_oscap.py [hunk 1]
```diff
@@ -599,6 +605,36 @@ def test_positive_associate_scap_policy_with_hostgroups(self, scap_content, modu
...
+                'hostgroups': hostgroup['name'],
+            }
+        )
+        assert scap_policy2['hostgroups'][0] == hostgroup['name']
+        module_target_sat.cli.HostGroup.delete({'id': hostgroup['id']})
+        # removal of hostgroup shouldn't affect policies
+        try:
+            module_target_sat.cli.Scappolicy.list()
+        except CLIReturnCodeError:
+            pytest.fail("failed to list policies")
```
Comment (Griffin-Sullivan) [inline tests/foreman/cli/test_oscap.py:626]: Why are we catching the failure?

**Review by Griffin-Sullivan — APPROVED** [review]

File: tests/foreman/cli/test_oscap.py [hunk 2]
```diff
@@ -599,6 +605,36 @@ def test_positive_associate_scap_policy_with_hostgroups(self, scap_content, modu
...
+                'period': OSCAP_PERIOD['weekly'].lower(),
+                'weekday': OSCAP_WEEKDAY['friday'].lower(),
+                'hostgroups': hostgroup['name'],
+            }
+        )
+        assert scap_policy2['hostgroups'][0] == hostgroup['name']
+        module_target_sat.cli.HostGroup.delete({'id': hostgroup['id']})
+        # removal of hostgroup shouldn't affect policies
+        try:
+            module_target_sat.cli.Scappolicy.list()
```
Comment (lhellebr) [inline tests/foreman/cli/test_oscap.py:624]: Shouldn't you assert the policies have indeed been listed?

File: tests/foreman/cli/test_oscap.py (hunk 1 above)
Comment (rplevka) [reply tests/foreman/cli/test_oscap.py:626]: I believe that plain raising of the exception results in `Error` junit status. If you properly fail it, you get `Failed` status.

**Review by rplevka — CHANGES_REQUESTED** [review]
ACK after adressing @lhellebr 's comment

File: tests/foreman/cli/test_oscap.py (hunk 1 above)
Comment (pnovotny) [reply tests/foreman/cli/test_oscap.py:626]: True, `CLIReturnCodeError` doesn't inherit from `AssertionError` so the test would result as `Error`. And unfortunately pytest doesn't have anything like `pytest.not_raises()` (yet), which would be cool for these cases.

**Comment by pnovotny** [comment]
I would also add some assertion like @lhellebr mentioned.
Otherwise LGTM.

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups
theforeman:
    foreman_openscap: 572

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups
theforeman:
    foreman_openscap: 572
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7012
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups --external-logging
Test Result : ========== 70 deselected, 76 warnings, 1 error in 1425.38s (0:23:45) ===========
```

**Comment by pondrejk** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups
theforeman:
    foreman_openscap: 572
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7015
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups --external-logging
Test Result : ========== 70 deselected, 76 warnings, 1 error in 1247.67s (0:20:47) ===========
```

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7124
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups --external-logging
Test Result : =========== 70 deselected, 78 warnings, 1 error in 717.56s (0:11:57) ===========
```

**Review by lhellebr — APPROVED** [review]

**Review by pnovotny — APPROVED** [review]

File: tests/foreman/cli/test_oscap.py [hunk 3]
```diff
@@ -572,16 +572,22 @@ def test_positive_associate_scap_policy_with_hostgroups(self, scap_content, modu
...
             2. Execute "policy" command with "create" as sub-command.
             3. Pass valid parameters.
-            4. Associate multiple hostgroups with policy
+            4. Associate multiple policies with hostgroup
+            5. Delete hostgroup
 
         :expectedresults: The policy is created and associated successfully.
+            Policies can be listed after hostgroup removal.
+
+        :bz: 1728157
```
Comment (jameerpathan111) [inline tests/foreman/cli/test_oscap.py:588]:
```suggestion
        :bz: 1728157
        
        :Verifies: SAT-19502
```
@pondrejk As we're moving away from Bugzilla, could we also add Jira issue here?

File: tests/foreman/cli/test_oscap.py [hunk 4]
```diff
@@ -599,6 +605,37 @@ def test_positive_associate_scap_policy_with_hostgroups(self, scap_content, modu
...
+                'weekday': OSCAP_WEEKDAY['friday'].lower(),
+                'hostgroups': hostgroup['name'],
+            }
+        )
+        assert scap_policy2['hostgroups'][0] == hostgroup['name']
+        module_target_sat.cli.HostGroup.delete({'id': hostgroup['id']})
+        # removal of hostgroup shouldn't affect policies
+        try:
+            result = module_target_sat.cli.Scappolicy.list()
+            assert name in [policy['name'] for policy in result]
```
Comment (ogajduse) [inline tests/foreman/cli/test_oscap.py:625]: Minor one: since CLIReturnCodeError can not be raised here, you can move this line below the except block.

**Review by jameerpathan111 — APPROVED** [review]

**Review by ogajduse — APPROVED** [review]
ACK, one minor comment left. 

co-reviewed with @jameerpathan111

File: tests/foreman/cli/test_oscap.py (hunk 3 above)
Comment (pondrejk) [reply tests/foreman/cli/test_oscap.py:587]: Added!

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 7357
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_oscap.py -k test_positive_associate_scap_policy_with_hostgroups --external-logging
Test Result : ========== 1 passed, 70 deselected, 79 warnings in 651.18s (0:10:51) ===========
```

**Pull request merged**